ocaml/xapi/xapi_vtpm.ml

```diff
@@ -0,0 +1,38 @@
+(*
+   Copyright (C) Citrix Systems Inc.
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU Lesser General Public License as published
+   by the Free Software Foundation; version 2.1 only. with the special
+   exception on linking described in file LICENSE.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU Lesser General Public License for more details.
+ *)
+
+let introduce ~__context ~uuid ~vM ~profile ~contents =
+  let ref = Ref.make () in
+  Db.VTPM.create ~__context ~ref ~uuid ~vM ~profile ~contents ;
+  ref
+
+let create ~__context ~vM =
+  let uuid = Uuid.(to_string (make_uuid ())) in
+  let profile = [] in
+  let contents = Xapi_secret.create ~__context ~value:"" ~other_config:[] in
+  let ref = introduce ~__context ~uuid ~vM ~profile ~contents in
+  ref
+
+let destroy ~__context ~self =
+  let secret = Db.VTPM.get_contents ~__context ~self in
+  Db.Secret.destroy ~__context ~self:secret ;
+  Db.VTPM.destroy ~__context ~self
+
+let get_contents ~__context ~self =
+  let secret = Db.VTPM.get_contents ~__context ~self in
+  Db.Secret.get_value ~__context ~self:secret
+
+let set_contents ~__context ~self ~contents =
+  let secret = Xapi_secret.create ~__context ~value:contents ~other_config:[] in
```

Comment:
Are we going to leak secrets here? I think we need to destroy the previous secret after calling set_contents below, otherwise we'll keep a growing list of secrets in the secrets table.